### DISCUSSION TITLE
Why lapis doesn't have support for other webservers?

### DISCUSSION THREAD

**Discussion by jodros** [leafo/lapis#778] [Ideas]
I want to  use Lapis with redbean (and maybe with any other webserver) instead of OpenResty, how to use lapis unbound to OpenResty?

I know this framework was develop with OpenResty in mind, but I think it'd be great to have a documented way to not depend only upon OpenResty.

**Comment by ryanford** [comment]
Lapis can already run independent from Openresty with a cqueues back-end. AFAIK those are the only two options right now

**Reply by jodros** [reply]
Glad to know about an alternative, but I never heard about this cqueues before, can you give me an example like a project you know that is using it?

**Reply by ryanford** [reply]
I have a few personal servers/websites using Lapis on Cqueues, works just fine. Only limitation I'm aware of is Leafo has mentioned DB connection pooling support is missing.

**Comment by Sudospective** [comment]
if both machines have access to unix sockets, you can run [this script i made](https://gist.github.com/Sudospective/c8f668870b2265bbdf800389ab2fb92e)
ill keep updating this, but it should be a quick and painless way to serve lapis apps. lemme know what its missing.

**Reply by Sudospective** [reply]
i should state that you might want to add a config to your nginx like so:
```
upstream laplace {
        server unix:/tmp/laplace.sock;
}

server {
        listen 80;
        listen [::]:80;
        server_name <server-name>;

        location / {
                proxy_pass http://laplace/;
                proxy_http_version 1.1;
                proxy_set_header Connection "";
                proxy_connect_timeout 60s;
                proxy_send_timeout 60s;
                proxy_read_timeout 60s;
                proxy_set_header Host $host;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Forwarded-Proto $scheme;

        }
}

```
youll use this server block to proxy to the socket, adjust your ssl stuff accordingly if hosting over the web

another thing to note is that this uses lua 5.4 (i needed a way to webhost lua 5.4 content), so if you dont have that, you might wanna get it or backport the code or help me do that

**Reply by Sudospective** [reply]
does anyone have more information about the request object and how to create one without going through ngx? i just need a function that gives me a proper request object i can use with `app.dispatch`.
